torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3575,6 +3575,47 @@ Tensor gather_with_keepdimed_indices(const Tensor& input, int64_t dim, const Ten
   return out_fw_grad;
 }
 
+Tensor attn_backward_v(const Tensor& output, const Tensor& tanh_output, const Tensor& grad_out_) {
+
+  Tensor grad_out = grad_out_;
+  if (!grad_out.defined()) {
+    grad_out = at::zeros(output.sizes(), at::kDouble);
```

Comment:
I think you want to optimize this code so that you skip doing computations altogether when the grad_output is not defined.

Reply:
Made changes based on this suggestion.